### PR TITLE
Feature/remove datepicker width property

### LINKED ISSUE #1894: Datepicker remove width property
As discussed, it isn't necessary to keep the width property because it only sets inline-styles to the component tag.
We can remove it.

Screenshot 2020-07-22 at 13 06 44

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1894

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Modifications still work in IE.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/20-molecules/datepicker/react/story.jsx [hunk 1]
```diff
@@ -41,14 +41,18 @@ storiesOf('Examples/Datepicker/React', module)
     const invalid = boolean('invalid', false);
     const placeholder = text('placeholder', 'Please select a date');
     const defaultValue = text('defaultValue', '');
-    const width = text('width', '400');
+    const inlineWidth = text('width (not an attribute)', '');
```
Comment (raphaellueckl) [inline src/components/20-molecules/datepicker/react/story.jsx:44]: What is this? Do we really need that?

File: src/utils/with-react.js [hunk 2]
```diff
@@ -142,7 +145,19 @@ const distributeProperties = (properties, componentClass) => {
 
     // map property name to value *unless* value is undefined
     if (value !== undefined) {
-      map[name] = value;
+      if (type === 'style') {
+        // {width: "500px"} -> width:500px;
```
Comment (raphaellueckl) [inline src/utils/with-react.js]: When I see stuff like that, I usually think an extra method, that explains what it does with its name, makes the code more readable.

File: src/components/20-molecules/datepicker/react/story.jsx (hunk 1 above)
Comment (raphaellueckl) [reply src/components/20-molecules/datepicker/react/story.jsx:44]: IMHO we should not have that as a prop. This is just general styling and we would somehow have to add it for all other elements too.

**Review by raphaellueckl — CHANGES_REQUESTED** [review]

File: src/components/20-molecules/datepicker/react/story.jsx (hunk 1 above)
Comment (01231) [reply src/components/20-molecules/datepicker/react/story.jsx:44]: It's nice to have for demo purposes in my opinion. And it's a real use case. We expect the user to use this feature in the datepicker.

**Comment by 01231** [comment]
seems like it doesn't work on ie

File: src/components/20-molecules/datepicker/react/story.jsx (hunk 1 above)
Comment (raphaellueckl) [reply src/components/20-molecules/datepicker/react/story.jsx:44]: A bit weird to me, but up to you.

File: src/utils/with-react.js (hunk 2 above)
Comment (01231) [reply src/utils/with-react.js]: done

**Review by raphaellueckl — APPROVED** [review]

**Pull request merged**